libs/utils/dummy_target.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class DummyTarget(object):
+    """
+    Dummy Devlib target
+    """
+
+    def __init__(self, real_target):
+        self._real_target = real_target
+        self._accu_cmds = []
+
+    @property
+    def accumulated_commands(self):
+        return self._accu_cmds
+
+    def clear_accumulator(self):
+        self._accu_cmds = []
+
+    # This is made to look like the devlib Target execute()
+    # Only the command argument is used, the rest is ignored
```

Comment:
> the rest is ignored

Then we should really throw an error if anything but the default value is provided. Also to be *really* correct we should start the script with `set -e` to get the correct `check_exit_code` behaviour.